### PR TITLE
Develop into master

### PR DESCRIPTION
**Pull request by rineless**

### CODE REVIEW

File: src/Application.java [hunk 1]
```diff
@@ -1,6 +1,80 @@
...
+import java.util.Locale;
+import java.util.stream.Collectors;
+
 public class Application {
 
     public static void main(String[] args){
-       System.out.println("Hello world");
+
+        System.out.println("Students:");
+        List<Student> studentList = receiveListOfStudents("src/data/StudentData.csv");
```
Comment (TetianaIefimenko) [inline src/Application.java]: try to use relative paths. After compilation you don't have such folder structure.

File: src/Application.java [hunk 2]
```diff
@@ -1,6 +1,80 @@
...
+        studentList.stream().forEach(student -> System.out.println(student.toString()));
+
+        System.out.println("Groups:");
+        List<Group> groupList = receiveListOfGroups("src/data/StudentGroupData.csv");
+        groupList.stream().forEach(group -> System.out.println(group.toString()));
+
+        List<String> studentsToAdd = new ArrayList<>();
+        studentsToAdd.add("13,Rubel,Bare,m,15.02.1998,German,Bremen,payable,0,present,rubel.bare@gmail.com");
```
Comment (TetianaIefimenko) [inline src/Application.java]: to add a student you need to construct a java object and convert it in a string in your code

File: src/Application.java [hunk 3]
```diff
@@ -1,6 +1,80 @@
...
+        groupsToAdd.add("4,Analysis,Anl,english,no,0,3,15,Karol Maier,karol.maier@myuni.de");
+        addGroup(groupsToAdd,groupList,"src/data/StudentGroupData.csv");*/
+
+    }
+
+    public static List<Student> receiveListOfStudents(String path){
+        FileReader studentReader = new FileReader();
+        List<String> lineList = studentReader.receiveLinesAsList(path);
+        CSVParser parser = new CSVParser();
+        lineList.remove(0);
```
Comment (TetianaIefimenko) [inline src/Application.java]: you can do this operation in the stream using `skip()`

File: src/model/group/Abbreviation.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package model.group;
+
+public enum Abbreviation {
```
Comment (TetianaIefimenko) [inline src/model/group/Abbreviation.java]: please, rename it `GroupNames` or give another descriptive name

File: src/model/group/Group.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+import java.util.Locale;
+
+public class Group {
+
+    private int id;
+    private String name;
+    private Abbreviation abbreviation;
+    private Locale language;
+    private boolean onlineAccess;
```
Comment (TetianaIefimenko) [inline src/model/group/Group.java]: boolean variables usually have names `is**`

File: src/util/parser/SeparatedValuesParser.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    private final String separator;
+
+    public SeparatedValuesParser(String separator_regex) {
+        this.separator = separator_regex;
+    }
+
+    public String[] parseLineToArray(String line) {
+        if (line != null)
+            return line.split(separator);
+        return null;
```
Comment (TetianaIefimenko) [inline src/util/parser/SeparatedValuesParser.java]: please, don't return null, return empty value or throw an exception

File: src/util/reader/FileReader.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import java.nio.file.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class FileReader extends Reader {
+
+    //TODO check exceptions, maybe make a bit different
+    public List<String> receiveLinesAsList(String path) {
+        try {
+            if (Files.exists(Path.of(path)) && Files.isReadable(Path.of(path)))
```
Comment (TetianaIefimenko) [inline src/util/reader/FileReader.java]: The path may specify a directory. Please, add an appropriate check for this case.

**Review by TetianaIefimenko — CHANGES_REQUESTED** [review]

File: src/Application.java (hunk 1 above)
Comment (rineless) [reply src/Application.java]: `Path current = Path.of(".");

        System.out.println("Students:");
        List<Student> studentList = receiveListOfStudents(current.toString() + "/src/data/StudentData.csv");
        studentList.stream().forEach(student -> System.out.println(student.toString()));

        System.out.println("Groups:");
        List<Group> groupList = receiveListOfGroups(current.toString() + "/src/data/StudentGroupData.csv");
        groupList.stream().forEach(group -> System.out.println(group.toString()));`

File: src/Application.java (hunk 2 above)
Comment (rineless) [reply src/Application.java]: `List<Student> studentsToAdd = new ArrayList<>();
        studentsToAdd.add(new Student().setId(13).setName("Rubel").setSurname("Bare").setGender(Gender.MALE)
                .setBirthDate(LocalDate.of(1998,2,15)).setCitizenship("German").setPlaceOfBirth("Bremen")
                .setTypeOfContract(TypeOfContract.PAYABLE).setGroupId(0).setTypeOfStudying(TypeOfStudying.PRESENT).setContactInformation("rubel.bare@gmail.com"));
        addStudents(studentsToAdd,studentList,current.toString() + "/src/data/StudentData.csv");`

File: src/Application.java (hunk 3 above)
Comment (rineless) [reply src/Application.java]: `FileReader studentReader = new FileReader();

        return studentReader.receiveLinesAsList(path).stream()
                .map(line -> new CSVParser().parseLineToStudent(line)).collect(Collectors.toList());`

File: src/model/group/Group.java (hunk 5 above)
Comment (rineless) [reply src/model/group/Group.java]: `private boolean isOnlineAccessible;`

File: src/model/group/Abbreviation.java (hunk 4 above)
Comment (rineless) [reply src/model/group/Abbreviation.java]: `public enum GroupNames`

File: src/util/parser/SeparatedValuesParser.java (hunk 6 above)
Comment (rineless) [reply src/util/parser/SeparatedValuesParser.java]: `public String[] parseLineToArray(String line) {
        if (line != null && !line.contentEquals("")) {
            return line.split(separator_regex);
        }
        return new String[]{};
    }`

File: src/util/reader/FileReader.java (hunk 7 above)
Comment (rineless) [reply src/util/reader/FileReader.java]: `public List<String> receiveLinesAsList(String path) {
        try {
            if (Files.exists(Path.of(path)) && Files.isRegularFile(Path.of(path))
                    && Files.isReadable(Path.of(path)))`

**Pull request merged**